pkg/controller/controller_utils.go

```diff
@@ -304,18 +304,30 @@ func filterActivePods(pods []api.Pod) []*api.Pod {
 // updateReplicaCount attempts to update the Status.Replicas of the given controller, with a single GET/PUT retry.
 func updateReplicaCount(rcClient client.ReplicationControllerInterface, controller api.ReplicationController, numReplicas int) (updateErr error) {
 	// This is the steady state. It happens when the rc doesn't have any expectations, since
-	// we do a periodic relist every 30s.
-	if controller.Status.Replicas == numReplicas {
+	// we do a periodic relist every 30s. If the sequenceNumbers differ but the replicas are
+	// the same, a caller might've resized to the same replica count.
+	if controller.Status.Replicas == numReplicas &&
+		controller.Spec.SequenceNumber == controller.Status.SequenceNumber {
 		return nil
 	}
+	// Save the sequence number we acted on, otherwise we might wrongfully indicate
+	// that we've seen a spec update when we retry.
+	// TODO: This can clobber an update if we allow multiple agents to write to the
+	// same status.
```

Comment:
Do we have a subresource for replication controller status yet? We really should do that.